webservice/src/main/java/org/daisy/pipeline/webservice/impl/JobResource.java

```diff
@@ -86,12 +90,33 @@ public Representation getResource() {
                 else {
                         writer = writer.withNewMessages(msgSeq);
                 }
+                if(job.get().getStatus() == Job.Status.IDLE){
+                        int pos = getPositionInQueue(job.get());
+                        writer.withQueuePosition(pos);
+                }
+
                 Document doc = writer.withScriptDetails().getXmlDocument();
                 
                 DomRepresentation dom = new DomRepresentation(MediaType.APPLICATION_XML, doc);
                 return dom;
         }
 
+        int getPositionInQueue(Job job){
+
+                Client client = webservice().getStorage().
+                        getClientStorage().defaultClient();
+                Collection<? extends Prioritizable<Job>> queue = webservice().getJobManager(client).getExecutionQueue().asCollection();
+                int pos = 0;
+                for (Prioritizable<Job> pJob: queue){
+                        pos++;
```

Comment:
Yes, as it's intended for the end user I though it would be more clear than using 0. The real deal for dealing with the queue for the client is through the queue api.